relay/kusama/src/ah_migration/phase1.rs

```diff
@@ -0,0 +1,177 @@
+	fn contains(call: &<Runtime as frame_system::Config>::RuntimeCall) -> bool {
+		let (_during, after) = call_allowed_status(call);
+		if !after {
+			log::warn!("Call bounced by the filter after the migration: {:?}", call);
+		}
+		after
+	}
+}
+
+/// The hold reason for staking delegation.
+pub struct StakingDelegationReason;
+impl Get<RuntimeHoldReason> for StakingDelegationReason {
+	fn get() -> RuntimeHoldReason {
+		RuntimeHoldReason::DelegatedStaking(pallet_delegated_staking::HoldReason::StakingDelegation)
+	}
+}
+
+/// Return whether a call should be enabled during and/or after the migration.
+///
+/// Time line of the migration looks like this:
+///
+/// --------|-----------|--------->
+///       Start        End
+///
+/// We now define 2 periods:
+///
+/// 1. During the migration: [Start, End]
+/// 2. After the migration: (End, ∞)
+///
+/// Visually:
+///
+/// ```text
+///         |-----1-----|
+///                      |---2---->
+/// --------|-----------|--------->
+///       Start        End
+/// ```
+///
+/// This call returns a 2-tuple to indicate whether a call is enabled during these periods.
+pub fn call_allowed_status(call: &<Runtime as frame_system::Config>::RuntimeCall) -> (bool, bool) {
+	use RuntimeCall::*;
+	const ON: bool = true;
+	const OFF: bool = false;
+
+	match call {
+		System(..) => (ON, ON), // Remarks, root calls and `set_code` if we need for emergency.
+		Scheduler(..) => (OFF, OFF), // Only for governance, hence disabled.
+		Preimage(..) => (OFF, OFF), // Only for governance, hence disabled.
+		Babe(..) => (ON, ON),   // For equivocation proof submissions; security relevant
+		Timestamp(..) => (ON, ON), // only `set` inherit
+		Indices(..) => (OFF, OFF), // Not needed anymore and migrated to AH.
+		Balances(..) => (OFF, ON), // Disabled during migration to avoid confusing externals.
+		Staking(..) => (OFF, OFF),
+		StakingAhClient(..) => (ON, ON), // Only permissioned calls and needed for the migration.
+		Session(..) => (ON, ON),         // Does not affect any migrating pallet.
+		Grandpa(..) => (ON, ON),         // For equivocation proof submissions; security relevant
+		Treasury(..) => (OFF, OFF),
+		ConvictionVoting(..) => (OFF, OFF),
+		Referenda(..) => (OFF, OFF),
+		Whitelist(..) => (OFF, OFF),
+		Claims(..) => (OFF, OFF),
+		Vesting(..) => (OFF, OFF),
+		Utility(..) => (ON, ON),   // batching etc
+		Proxy(..) => (OFF, ON),    // On after the migration to keep proxy accounts accessible.
+		Multisig(..) => (OFF, ON), // On after the migration to keep multisig accounts accessible.
+		Bounties(..) => (OFF, OFF),
+		ChildBounties(..) => (OFF, OFF),
+		ElectionProviderMultiPhase(..) => (OFF, OFF),
+		VoterList(..) => (OFF, OFF),
+		NominationPools(..) => (OFF, OFF),
+		FastUnstake(..) => (OFF, OFF),
+		Configuration(..) => (ON, ON), /* TODO allow this to be called by fellow origin during the migration https://github.com/polkadot-fellows/runtimes/pull/559#discussion_r1928794490 */
```

Comment:
remove or address?

Reply:
Removed now, we dont have the time for this.